src/Compilers/CSharp/Test/Semantic/SourceGeneration/GeneratorDriverTests_Attributes_FullyQualifiedName.cs

```diff
@@ -1304,6 +1304,53 @@ static void replace(ref Compilation compilation, CSharpParseOptions parseOptions
         }
     }
 
+    [Fact, WorkItem("https://github.com/dotnet/roslyn/pull/70791")]
+    public void FindCorrectAttributeOnTopLevelClass_WhenSearchingForClassDeclaration1_DeepRecursion()
+    {
+        var deeplyRecursive = string.Join("+", Enumerable.Repeat(""" "a" """, 20_000));
```

Comment:
test fails at around 11,000 frames on my machine.

Reply:
Should this go in EndToEndTests.cs, with the rest of the deeply-recursive tests?

Reply:
Moved test.